testnet/stacks-node/src/mockamoto.rs

```diff
@@ -377,23 +806,57 @@ impl MockamotoNode {
             tenure_change_tx_payload,
         );
         tenure_tx.chain_id = chain_id;
-        tenure_tx.set_origin_nonce(miner_nonce + 1);
-        let txid = tenure_tx.txid();
+        tenure_tx.set_origin_nonce(miner_nonce);
         let mut tenure_tx_signer = StacksTransactionSigner::new(&tenure_tx);
         tenure_tx_signer.sign_origin(&self.miner_key).unwrap();
         let tenure_tx = tenure_tx_signer.get_tx().unwrap();
 
+        let pox_address = PoxAddress::Standard(
+            StacksAddress::burn_address(false),
+            Some(AddressHashMode::SerializeP2PKH),
+        );
+
+        let stack_stx_payload = if parent_chain_length < 2 {
+            TransactionPayload::ContractCall(TransactionContractCall {
+                address: StacksAddress::burn_address(false),
+                contract_name: "pox-4".try_into().unwrap(),
+                function_name: "stack-stx".try_into().unwrap(),
+                function_args: vec![
+                    ClarityValue::UInt(99_000_000_000_000),
+                    pox_address.as_clarity_tuple().unwrap().into(),
+                    ClarityValue::UInt(u128::from(parent_burn_height)),
+                    ClarityValue::UInt(12),
+                ],
+            })
+        } else {
+            // NOTE: stack-extend doesn't currently work, because the PoX-4 lockup
```

Comment:
Doesn't this block this PR, then?

Reply:
Not in my opinion -- the mockamoto node runs fine until it reaches 12 reward cycles, and there's very little overlap between the PoX-4 lockup work and the work in this PR, so I think they should be considered separately.

Reply:
kk